File: 3_Arrays/Sortings/pancake sort/pancake sort.js
function flip(array, len) {
    for (let i=0, j=len; i<j; i++, j--) {
        let temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }
}

function pancakeSort(array) {
    for (let len = array.length; len > 1; --len) { 
        let max = array.indexOf(Math.max(...array.slice(0, len)));
        if (max != len-1) { 
            flip(array, max); 
            flip(array, len-1); 
        }
    }
}


let arr = new Array(9)
for (let i = 0; i < arr.length; i++) 
    arr[i] = Math.floor(Math.random() * 10) 


console.log(arr)
pancakeSort(arr);
console.log(arr)

File: 3_Arrays/Sortings/pancake sort/pancake_sort.py
from random import randint

def pancake_sort(lst):
    for i in range(len(lst)-1, 0, -1):
        k = lst[0:i+1].index(max(lst[0:i+1]))
        print(k)
        if k != i:
            lst = lst[k::-1]+lst[k+1::]
            print(lst)
            lst = lst[i::-1]+lst[i+1::]
        print(lst)
    return lst

a = [randint(0,20) for i in range(10)]
print(a)
print(pancake_sort(a))
